### ISSUE TITLE
Compiler warning in Http class

### ISSUE THREAD

**Issue by b0guslaw** [SFML/SFML#2217]
## Wrestrict triggered in Http.cpp

The use of `std::string::insert()` in Http.cpp:32 causes the compiler warning -Wrestrict to trigger. This was noticed in a C++20 environment.

## Your environment

* EndeavourOS with gcc 12.2.0
* 2.6.x but occurs on all branches including master
* Set language standard to C++20
* -Wall -Werror -Wpedantic

## Steps to reproduce

Simply build a minimal SFML example from source

## Expected behavior

No warnings should be triggered while building SFML

## Actual behavior

```
inlined from ‘void sf::Http::Request::setUri(const std::string&)’ at <snip>/build/_deps/sfml-src/src/SFML/Network/Http.cpp:82:21:
/usr/include/c++/12.2.0/bits/char_traits.h:431:56: error: ‘void* __builtin_memcpy(void*, const void*, long unsigned int)’ accessing 9223372036854775810 or more bytes at offsets [2, 9223372036854775807] and 1 may overlap up to 9223372036854775813 bytes at offset -3 [-Werror=restrict]
  431 |         return static_cast<char_type*>(__builtin_memcpy(__s1, __s2, __n));
      |                                        ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [_deps/sfml-build/src/SFML/Network/CMakeFiles/sfml-network.dir/build.make:90: _deps/sfml-build/src/SFML/Network/CMakeFiles/sfml-network.dir/Http.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:389: _deps/sfml-build/src/SFML/Network/CMakeFiles/sfml-network.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**Comment by kimci86** [comment]
This is a known bug in GCC when using C++20 and optimizations. See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=105651

**Comment by ChrisThrasher** [comment]
I think the patch in #2218 is a reasonable workaround. I want people using cutting edge compilers with warnings enabled. `std::string` provides 10 redundant APIs for every operation so it's an easy to switch to an iterator-based API instead. Plus I think it makes the code marginally more expressive which is nice.

**Comment by eXpl0it3r** [comment]
Fixed with #2218
